pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -117,90 +121,194 @@ func (n *NetworkPolicyController) deleteCNP(old interface{}) {
 	n.deleteDereferencedAddressGroups(oldInternalNP)
 }
 
+// filterPerNamespaceRuleACNPsByNSLabels gets all ClusterNetworkPolicy names that will need to be
+// re-processed if a Namespace adds or removes the input labels.
+func (n *NetworkPolicyController) filterPerNamespaceRuleACNPsByNSLabels(nsLabels labels.Set) sets.String {
+	n.internalNetworkPolicyMutex.Lock()
+	defer n.internalNetworkPolicyMutex.Unlock()
+
+	affectedPolicies := sets.String{}
+	nps, err := n.internalNetworkPolicyStore.GetByIndex(store.PerNamespaceRuleIndex, store.HasPerNamespaceRule)
+	if err != nil {
+		klog.Errorf("Error fetching internal NetworkPolicies that have per-Namespace rules: %v", err)
+		return affectedPolicies
+	}
+	for _, np := range nps {
+		internalNP := np.(*antreatypes.NetworkPolicy)
+		//klog.Infof("NP %v has perNSSel", internalNP.SourceRef.Name)
+		for _, sel := range internalNP.PerNamespaceSelectors {
```

Comment:
I think that would be difficult. The appliedToGroup splitting by Namespace takes place in ACNP processing, and the resulting appliedToGroups do not have information on the Namespace selectors, so `GetGroupsForNamespace` won't be able to filter out those groups. If we want to use the Group Entity index, then we should make the index aware of policies in addition to groups, which I don't feel is ideal

Reply:
> If we want to use the Group Entity index, then we should make the index aware of policies in addition to groups, which I don't feel is ideal

It's not in addition to Groups, it's in addition to Pods and ExternalEntities, which I agree is not ideal. In theory, I think that the Group Entity index can support arbitrary logical groups, it could be groups of Pods, or groups of internal policies. But the amount of changes necessary indeed does not seem worth it.